README.md

[![Lifecycle:
stable](https://img.shields.io/badge/lifecycle-stable-brightgreen.svg)](https://www.tidyverse.org/lifecycle/#maturing)
[![](https://img.shields.io/github/last-commit/mattwarkentin/CAIC-RT.svg)](https://github.com/mattwarkentin/CAIC-RT/commits/master)
![GitHub repo
size](https://img.shields.io/github/repo-size/mattwarkentin/CAIC-RT)
[![Licence](https://img.shields.io/github/license/mashape/apistatus.svg)](http://choosealicense.com/licenses/mit/)

### CAIC-RT: COVID-19 Acute and Intensive Care Resource Tool

This repository contains all of the code for the
[**CAIC-RT**](https://caic-rt.shinyapps.io/CAIC-RT/?utm_source=github&utm_medium=readme)
online tool.

-----

Please note that the ‘CAIC-RT’ project is released with a [Contributor
Code of Conduct](CODE_OF_CONDUCT.md). By contributing to this project,
you agree to abide by its terms.


NEWS.md

# CAIC-RT

------------------------------------------------------------------------

## v1.2.0 (2021-01-04)

-   Updated length of stay, percent of critical care patients requiring
    mechanical ventilation, and age-specific information based on
    emerging Canadian data, [Mac et al.,
    2020](https://www.medrxiv.org/content/10.1101/2020.11.04.20225474v1)
    -   Length of stay (in days): Acute care \[11 → 11.8\], Critical
        care \[9 → 16.4\], Mechanical ventilator \[9 → 20.5\]
    -   Percent of critical care patients requiring mechanical
        ventilation \[70% → 60%\]
    -   Finer age-specific strata with updated case distribution and
        severity data

------------------------------------------------------------------------

## v1.1.0 (2020-04-07)

### Major Changes

-   Default length of stay in Critical Care and duration of use for
    Mechanical Ventilators have been updated to **9 days** from **20
    days** based on emegring data from Italy, [Grasselli et
    al. (2020)](https://jamanetwork.com/journals/jama/fullarticle/2764365?utm_source=twitter).

-   Default percent of critical care patients requiring mechanical
    ventilation has been updated to **70%** from **50%** based on
    emerging data from Italy, [Grasselli et
    al. (2020)](https://jamanetwork.com/journals/jama/fullarticle/2764365?utm_source=twitter).

    -   The slider has been moved from its original location to within
        the collapsible panel containing the age-stratified case
        distribution data.

### New Features

-   Age-stratified case distribution and severity table now supports
    users uploading CSV or Excel files with custom data

-   Multilingual: CAIC-RT now supports French and Spanish language
    translations, in addition to English (\#2).

-   Walkthrough: Added a tutorial video explaining how to use and
    interpret CAIC-RT

-   Added a bookmark button to allow users to generate a URL that
    captures the current state of the inputs

    -   The URL allows users to return to the tool with the same inputs
        used previously.

------------------------------------------------------------------------

## v1.0.2 (2020-03-28)

-   PDF report generation feature was added to tool

------------------------------------------------------------------------

## v1.0.1 (2020-03-25)

-   Minor bug fix for updating the feedback warning generated when the
    number of critical care beds was less than the number of mechanical
    ventilators
    -   Previously, if the feedback warning was triggered when changing
        the critical care input, the feedback warning was not properly
        removed when the mechanical ventilator input was updated (and
        vice versa).

------------------------------------------------------------------------

## v1.0.0 (2020-03-24)

-   Initial public tool deployment


app/ui.R
# Load packages ----
source('package-deps.R')

# Load functions ----
source('utils.R')

# Meta Content ----
tool_meta <-
  meta() %>%
  meta_general(
    application_name = "CAIC-RT",
    description = "An online tool capable of estimating the maximum daily number of incident COVID-19 cases manageable by healthcare systems."
  ) %>%
  meta_social(
    title = "CAIC-RT - COVID-19 Acute and Intensive Care Resource Tool",
    description = "An online tool capable of estimating the maximum daily number of incident COVID-19 cases manageable by healthcare systems.",
    url = "https://caic-rt.shinyapps.io/CAIC-RT",
    image = "https://images.unsplash.com/photo-1584118624012-df056829fbd0?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1778&q=80",
    image_alt = "Image of the coronavirus courtesy of the CDC",
    twitter_card_type = "summary",
    og_author = c('Vasily Giannakeas', 'Deepit Bhatia',
                  'Matthew T. Warkentin', 'Isaac I. Bogoch', 'Nathan M. Stall')
  )

# Navbar UI ----

# Based on ISO 639-2/B
# https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes
langs <- sort(c('English' = 'eng', 'Spanish' = 'spa',
                'French' = 'fre'))

ui <- function(request) {
  navbarPage(
    # Navbar ----
    title = 'CAIC-RT',
    footer = htmlOutput('dev_by'),
    inverse = TRUE,
    fluid = TRUE,
    position = 'fixed-top',
    collapsible = TRUE,
    header = tags$style(type="text/css", "body {padding-top: 70px;}"),
    # Home Tool ----
    tabPanel(
      htmlOutput('home'),
      useShinyalert(),
      useShinyFeedback(),
      tool_meta,
      tags$head(
        tags$link(rel = "stylesheet", type = "text/css",
                  href = "styles.css"),
        tags$link(rel = "stylesheet", type = "text/css",
                  href = "tachyons.min.css"),
        includeHTML('www/google-analytics.html')
      ),

      fluidRow(

        # First Column ----
        column(3,
               htmlOutput('last_update'),
               selectInput('lang', 'Choose Language',
                           choices = langs,
                           selected = 'English',
                           multiple = FALSE),
            actionButton('about_tool', ' About This Tool', icon = icon('info'),
                         class = "mb3 btn-primary f4 f5-ns small-caps")
            ,
            actionButton('contribute', ' How can I contribute?',
                         icon = icon('hands-helping'),
                         class = 'f4 f5-ns bg-gold hover-bg-yellow small-caps mb3'),
               wellPanel(
               htmlOutput('p1_header'),
               numericInput(inputId = "lou_acute",
                            label = htmlOutput('lou_acute_label'),
                            value = 11.8,
                            min = 1),

               numericInput(inputId = "lou_crit",
                            label = htmlOutput('lou_crit_label'),
                            value = 16.4, min = 1),

               numericInput(inputId = "lou_vent",
                            label = htmlOutput('lou_vent_label'),
                            value = 20.5, min = 1),

               htmlOutput('p1_footnote'),
               )
               ),

        # Second Column ----
        column(4,
               wellPanel(
               htmlOutput('p2_header'),

               numericInput(inputId = "n_acute",
                            label = htmlOutput('n_acute_label'),
                            value = 8378, min = 0),
               actionButton('acute', 'Calculate Acute Care Beds',
                            icon = icon('cogs'),
                            class = 'bg-dark-gray white f5 f5-l f7-m'),
               hr(),

               numericInput(inputId = "n_crit",
                            label = htmlOutput('n_crit_label'),
                            value = 513, min = 0),
               actionButton('critical', 'Calculate Critical Care Beds',
                            icon = icon('cogs'),
                            class = 'bg-dark-gray white f5 f5-l f7-m'),
               hr(),

               numericInput(inputId = "n_vent",
                            label = htmlOutput('n_vent_label'),
                            value = 328, min = 0),
               actionButton('mvent', 'Calculate Mechanical Ventilators',
                            icon = icon('cogs'),
                            class = "bg-dark-gray white f5 f5-l f7-m mb4"),

               htmlOutput('p2_footnote_2')
               ), bsCollapse(id = 'global', multiple = TRUE,
                             bsCollapsePanel(
                               title = htmlOutput('table_title'),
                               uiOutput('input_data'),
                               DT::dataTableOutput("tab_pop"),
                               htmlOutput('table_source'),
                               htmlOutput('table_tip'),
                               actionButton('reset', 'Reset', class = 'btn-warning f4 f4-l f5-m',
                                            icon = icon('redo')),
                               hr(),
                               sliderInput(inputId = "per_vent",
                                           label = htmlOutput('per_vent_label'),
                                           value = 60, min = 0, max = 100, step = 1,
                                           post = '%'),
                               htmlOutput('p2_footnote_1'),
                               style = 'primary'
                             ), open = ''
               )),

        # Third Column ----
        column(5,
               htmlOutput('plot_title'),
        htmlOutput('plot_desc'),

        span(dropdownButton(size = 'xs', icon = icon('cog'),
                            tooltip = TRUE, inline = TRUE, width = '200px',
                            selectInput('colors', 'Choose Colour Palette',
                                        choices = c('YlOrRd', 'Viridis',
                                                    'Grayscale', 'Pastel',
                                                    'Brewer', 'Default'),
                                        selected = 'YlOrRd',
                                        multiple = FALSE)),
             plotlyOutput('plot')),
        wellPanel(
          htmlOutput('interpret_title'),
          tabsetPanel(id = 'interpretations',
                      tabPanel(htmlOutput('acute_res_title'),
                               br(),
                               htmlOutput('acute_int')),
                      tabPanel(htmlOutput('crit_res_title'),
                               br(),
                               htmlOutput('crit_int')),
                      tabPanel(htmlOutput('vent_res_title'),
                               br(),
                               htmlOutput('mv_int'))),
          hr(),
          div(uiOutput('report', class = 'dib'),
               uiOutput('bookmark', class = 'dib'),
              class = 'tc')
          )
            )
      )
      ),

    # Help Page ----
    tabPanel(
      htmlOutput('help'),
      htmlOutput('help_text')
    ),
    # What's New? ----
    tabPanel(htmlOutput('whats_new'),
             htmlOutput('whats_new_text')),

    # Tutorial ----
    tabPanel(htmlOutput('tutorial'),
             HTML('<iframe width="560" height="315" src="https://www.youtube.com/embed/owjI123tUrE" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>')),

    # More Info ----
    navbarMenu(htmlOutput('more_info', inline = TRUE),
               tabPanel(tags$a(htmlOutput('see_article'), href = "https://www.acpjournals.org/doi/10.7326/M20-1169", target = "_blank")),
               tabPanel(tags$a(htmlOutput('see_code'), href = "https://www.github.com/mattwarkentin/CAIC-RT", target = "_blank"))
    )
  )
}
